apps/console/src/features/api-resources/components/api-resource-panes/general-api-resource.tsx

```diff
@@ -0,0 +1,312 @@
+} from "../../models";
+
+/**
+ * Prop-types for the API resources page component.
+ */
+type GeneralAPIResourceInterface = SBACInterface<FeatureConfigInterface> & 
+    IdentifiableComponentInterface & APIResourcePanesCommonPropsInterface;
+
+const FORM_ID: string = "apiResource-general-details";
+
+/**
+ * API Resources listing page.
+ *
+ * @param props - Props injected to the component.
+ * @returns API Resources Page component
+ */
+export const GeneralAPIResource: FunctionComponent<GeneralAPIResourceInterface> = (
+    props: GeneralAPIResourceInterface
+): ReactElement => {
+
+    const {
+        isReadOnly,
+        isSubmitting,
+        handleUpdateAPIResource,
+        apiResourceData,
+        isAPIResourceDataLoading,
+        ["data-componentid"]: componentId
+    } = props;
+
+    const { t } = useTranslation();
+    const dispatch: Dispatch = useDispatch();
+
+    const [ isFormValidationError, setIsFormValidationError ] = useState<boolean>(true);
+    const [ deleteAPIResourceLoading, setDeleteAPIResourceLoading ] = useState<boolean>(false);
+    const [ showDeleteConfirmationModal, setShowDeleteConfirmationModal ] = useState<boolean>(false);
+    const [ deletingAPIResource, setDeletingAPIResource ] = useState<APIResourceInterface>(undefined);
+
+    /**
+     * Deletes an API resource when the delete API resource button is clicked.
+     *
+     * @param apiResourceId - API resource ID.
+     */
+    const handleAPIResourceDelete = (apiResourceId: string): void => {
+
+        setDeleteAPIResourceLoading(true);
+        deleteAPIResource(apiResourceId)
+            .then(() => {
+                dispatch(addAlert<AlertInterface>({
+                    description: t("extensions:develop.apiResource.notifications.deleteAPIResource.success" +
+                        ".description"),
+                    level: AlertLevels.SUCCESS,
+                    message: t("extensions:develop.apiResource.notifications.deleteAPIResource.success.message")
+                }));
+
+                setShowDeleteConfirmationModal(false);
+                history.push(APIResourcesConstants.getPaths().get("API_RESOURCES"));
+            })
+            .catch(() => {
+                dispatch(addAlert<AlertInterface>({
+                    description: t("extensions:develop.apiResource.notifications.deleteAPIResource" +
+                        ".genericError.description"),
+                    level: AlertLevels.ERROR,
+                    message: t("extensions:develop.apiResource.notifications.deleteAPIResource" +
+                        ".genericError.message")
+                }));
+            })
+            .finally(() => {
+                setDeleteAPIResourceLoading(false);
+                setDeletingAPIResource(undefined);
+            });
+    };
+
+    /**
+     * Show danger zone component
+     * 
+     * @returns `ReactElement`
+     */
+    const resolveDangerActions = (): ReactElement => {
+        return (
```

Comment:
Lets improve readability.
```suggestion
if (isReadOnly) {
    return null;
}

return (
    <Show
        // rest
```